apidump/apidump.go

```diff
@@ -457,10 +472,20 @@ func Run(args Args) error {
 		}
 		printer.Stderr.Infof("Running learn mode on interfaces %s\n", strings.Join(iNames, ", "))
 	}
-	if len(outboundFilters) == 0 {
+
+	unfiltered := true
+	for _, f := range inboundFilters {
+		if f != "" {
+			unfiltered = false
+			break
+		}
+	}
+	if unfiltered {
 		printer.Stderr.Warningf("%s\n", printer.Color.Yellow("--filter flag is not set, this means that:"))
 		printer.Stderr.Warningf("%s\n", printer.Color.Yellow("  - all network traffic is treated as your API traffic"))
-		printer.Stderr.Warningf("%s\n", printer.Color.Yellow("  - outbound witness collection is disabled"))
+		if args.EnableOutbound {
+			printer.Stderr.Warningf("%s\n", printer.Color.Yellow("  - outbound witness collection is disabled"))
+		}
 	}
 
 	var stopErr error
```

Comment:
I think we need to change this (below the window of your changes) based on EnableOutbound?  It works when `--debug` is enabled, but the check for 0 is inappropriate otherwise.

```
			if outboundSummary.Total().TCPPackets == 0 {
				printer.Stderr.Infof("Did not capture any TCP packets during the trace.\n")
				printer.Stderr.Infof("%s\n", printer.Color.Yellow("This may mean the traffic is on a different interface, or that"))
				printer.Stderr.Infof("%s\n", printer.Color.Yellow("there is a problem sending traffic to the API."))
			} else {
				printer.Stderr.Infof("Did not capture any TCP packets matching the filter.\n")
				printer.Stderr.Infof("%s\n", printer.Color.Yellow("This may mean your filter is incorrect, such as the wrong TCP port."))
			}
```

Reply:
Fixed in 5e0a89a to also guard on `capturingOutbound`.